pkg/apis/core/validation/validation.go

```diff
@@ -3905,42 +3907,57 @@ func validateSysctls(sysctls []core.Sysctl, fldPath *field.Path) field.ErrorList
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("runAsUser"), *(securityContext.RunAsUser), msg))
+			if securityContext.RunAsUser != nil {
+				for _, msg := range validation.IsValidUserID(*securityContext.RunAsUser) {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("runAsUser"), *(securityContext.RunAsUser), msg))
+				}
 			}
-		}
-		if securityContext.RunAsGroup != nil {
-			for _, msg := range validation.IsValidGroupID(*securityContext.RunAsGroup) {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("runAsGroup"), *(securityContext.RunAsGroup), msg))
+			if securityContext.RunAsGroup != nil {
+				for _, msg := range validation.IsValidGroupID(*securityContext.RunAsGroup) {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("runAsGroup"), *(securityContext.RunAsGroup), msg))
+				}
 			}
-		}
-		for g, gid := range securityContext.SupplementalGroups {
-			for _, msg := range validation.IsValidGroupID(gid) {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("supplementalGroups").Index(g), gid, msg))
+			for g, gid := range securityContext.SupplementalGroups {
+				for _, msg := range validation.IsValidGroupID(gid) {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("supplementalGroups").Index(g), gid, msg))
+				}
+			}
+			if securityContext.ShareProcessNamespace != nil && securityContext.HostPID && *securityContext.ShareProcessNamespace {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("shareProcessNamespace"), *securityContext.ShareProcessNamespace, "ShareProcessNamespace and HostPID cannot both be enabled"))
 			}
-		}
-		if securityContext.ShareProcessNamespace != nil && securityContext.HostPID && *securityContext.ShareProcessNamespace {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("shareProcessNamespace"), *securityContext.ShareProcessNamespace, "ShareProcessNamespace and HostPID cannot both be enabled"))
-		}
 
-		if len(securityContext.Sysctls) != 0 {
-			allErrs = append(allErrs, validateSysctls(securityContext.Sysctls, fldPath.Child("sysctls"))...)
-		}
+			if len(securityContext.Sysctls) != 0 {
+				allErrs = append(allErrs, validateSysctls(securityContext.Sysctls, fldPath.Child("sysctls"))...)
+			}
 
-		if securityContext.FSGroupChangePolicy != nil {
-			allErrs = append(allErrs, validateFSGroupChangePolicy(securityContext.FSGroupChangePolicy, fldPath.Child("fsGroupChangePolicy"))...)
+			if securityContext.FSGroupChangePolicy != nil {
+				allErrs = append(allErrs, validateFSGroupChangePolicy(securityContext.FSGroupChangePolicy, fldPath.Child("fsGroupChangePolicy"))...)
+			}
+			allErrs = append(allErrs, validateSeccompProfileField(securityContext.SeccompProfile, fldPath.Child("seccompProfile"))...)
+		}
+		// OS based podSecurityContext validation
+		// TODO: Think if we need to relax this restriction or some of the restrictions
+		if os != nil && os.Name == "windows" {
```

Comment:
Should we add these tot he list?
- runAsUser
- runAsGroup
- supplementalGroups
- privileged
- allowPrivilegeEscalation
- capabilities (both add / drop)

---

Need to verify if these work work on Windows or not
- readOnlyRootFilesystem (I don't think it does)
- procMount

----

Also note that `runAsNonRoot` IS supported for Windows.